### DISCUSSION TITLE
esp32 with AhoyDTU stops working

### DISCUSSION THREAD

**Discussion by michaels-github** [lumapu/ahoy#1793] [General]
Hi, since some weeks sometimes my AhoyDTU stop working (after some days). I can not ping the device and connect. Only power off works.
After power off/on it  becomes online again and send mqtt messages.
Is there a way to connect to the esp23 for trouble shooting?
Best regards,
Michael

Workaround: I created a cronjob on wireless router. If ping is not answered it will send a mqtt message to server and 
power off/on via shelly device

**Comment by schnuller123** [comment]
Hallo
ich habe ähnliches Problem!
ESP32 ohne Display nur mit NRF24L01 und aktuelle Firmware
eingebunden mit MQTT, WiFi (SSID: Terasse, RSSI: -69, CH: 2)
Wlan von einer DLAN Strecke!
abstecken und autoneustart und er geht wieder

läuft manchmal 3 wochen ohne Probleme, dann wieder nur 1 Woche
passiert also nicht sehr häufig!

kann mann da nicht einen Watchdog reinmachen der neustart auslöst, wenn kein Wlanverbindung mehr besteht

überlege schon einen Wlansteckdose anzusteuern wenn keine Daten kommen
ist aber schon viel aufwand!!!

**Comment by schnuller123** [comment]
scheinbar kein interresse einen wlan watchdog einzubauen
schade!!!!!

**Reply by stefan123t** [reply]
@schnuller123 bitte USB-Serial Logs von dem Zeitraum in dem der Disconnect und kein Reconnect auftreten in einem Issue beisteuern, nur so können wir analysieren was genau die Ursache des Problems bei Dir ist. Da es sich um einen Verbindungsabbruch der WLAN Verbindung handelt, geht das nur mit USB-Serial Monitor / Logs.

**Reply by schnuller123** [reply]
@stefan123t 
anbei ein log der Fehler/Neustart und paar merkwürdige sachen
aber leider kein dauerhafter Stop ohne WLAN ( passiert nur 1x in der woche )
[log_nacht mit fehler.txt](https://github.com/user-attachments/files/20637424/log_nacht.mit.fehler.txt)
vielleicht findest du da was raus

**Reply by stefan123t** [reply]
Hallo @schnuller123 
also Watchdog ist doch einer da der hat den ESP ja neugestartet.

Ich kann in dem Log leider nicht sehr viel erkennen ausser dass der loopTask zu lange (> 1100-1300ms) braucht und den Watchdog nicht rechtzeitig wieder schlafen legt.

1. An der Stromversorgung liegt es nicht ? ESP32 Abstürze können evtl. auch durch ungenügende Spannungsversorgung ausgelöst werden. Der Kondensator ist dran ?

2. Du versuchst die ganze Nacht dem WR irgendwelche Power Limits zu senden, warum ? Der hat doch keinen PV Strom und antwortet daher nicht.

3. Die Logs unmittelbar vor den beiden Abstürzen fehlen. Evtl gibt es ja einen Hinweis warum der ESP den Watchdog nicht füttert ? Was macht er unmittelbar davor ?

4. zumindest beim ersten Absturz scheint der WiFiClient.cpp in Zeile 429 eine Verbindung nicht mehr zu bekommen. Wie stabil ist denn diese DLAN Verbindung ?

5. Der MQTT Task versucht noch 20 Sekunden lang die Verbindung zu halten, bis ihm der Watchdog in die Waden beißt. Evtl klappt hier etwas beim Abbau / Neuaufbau der WiFi Verbindung nicht ganz.

ME kann die WiFi Bibliothek nicht alle Disconnect reasons sauber abfangen und/oder die MQTT Bibliothek kann die TCP Verbindung darüber nicht mehr aufbauen.

Da gibt es mW auch ein Ticket dazu. Wo haben wir das bloß hin ?

Ach ja unter anderem hier:

https://github.com/lumapu/ahoy/issues/1765#issuecomment-2889042441

```
6:07:05.208 -> [44806157][E][WiFiClient.cpp:429] write(): fail on fd 48, errno: 104, "Connection reset by peer"
06:07:05.208 -> I: MQTT disconnected, reason: TCP disconnect
06:07:06.204 -> I: MQTT connected
06:07:26.666 -> E (44844778) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
06:07:26.666 -> E (44844778) task_wdt:  - loopTask (CPU 1)
06:07:26.666 -> E (44844778) task_wdt: Tasks currently running:
06:07:26.725 -> E (44844778) task_wdt: CPU 0: wifi
06:07:26.725 -> E (44844778) task_wdt: CPU 1: IDLE1
06:07:26.725 -> E (44844778) task_wdt: Aborting.
06:07:26.725 -> 
06:07:26.725 -> abort() was called at PC 0x401186e0 on core 0
```

**Comment by stefan123t** [comment]
@lumapu in OpenDTU sind wir evtl. schon einen kleinen Schritt weiter bei der Analyse des Problems:
Dort tritt immer wieder ein "disconnect reason" `2 WIFI_REASON_AUTH_EXPIRE` als Ursache des Verbindungsverlustes auf.

Dieser "Verbindungsabbruch" wird von Arduino/ESP32 mit einem Retry und nicht mit einem Reconnect behandelt.
Das scheint mit Unterschieden in der Behandlung des Disconnects zwischen Arduino/ESP32 und ESP-IDF zusammen zu hängen.

https://github.com/tbnobody/OpenDTU/issues/2202#issuecomment-2888615142

Ich kenne den aktuellen AhoyDTU Code zum Verbindungsaufbau aktuell nicht gut genug,
aber der Code den @jstammi hier in AquaMQTT gesichtet hat, klingt für mich relativ einleuchtend.

https://github.com/tbnobody/OpenDTU/issues/2400#issuecomment-2881784534

**Reply by stefan123t** [reply]
Siehe auch hier [#1765 Ahoy DTU verliert permanent WLAN Verbindung bei TP Link Repeater](https://github.com/lumapu/ahoy/issues/1765#issuecomment-2903429391)

**Comment by doetnetalsof** [comment]
Gleiches Problem hier, vom ersten Tag an, als ich das aho-dtu gekauft habe, war es eine Art Russisches Roulette, ob am nächsten Tag eine WLAN-Verbindung bestehen würde oder nicht, selbst der automatische Reset macht keinen Unterschied, was immer funktioniert, ist Aus-/Einschalten, aber nach einer Weile macht man das nicht mehr.

Gibt es schon einen Hinweis, wann das nächste Update kommt, v0.8.155 ist ja schon eine Weile alt ;-)

Keep up the good work  ;-)

**Reply by michaels-github** [reply]
Hallo, ich nutzte nun richtiges Raspberry PI  Netzteil mit gutem USB-Kabel und habe parrallel zur Stromversorgung einen Kondensator geschaltet. Seit dem läuft der esp32  wesentlich stabiler. 
Möglicherwiese sorgt das Funkmodul (NRF24L01+) beim  Aktieren für einen starken Spannungsabfall und der esp32 will nicht mehr.

https://arduino-hannover.de/2018/07/25/die-tuecken-der-esp32-stromversorgung/

**Comment by oremountains** [comment]
Das Problem hatte ich auch hab dann Elko 470uF direkt an das NRF24L01+ gelötet parallel dazu 100nF Tantal und je 100nF Tantal an den ESP an +5V und +3,3V dann war Ruhe.
